src/components/Planning/GroupDialog/GroupDialog.tsx

```diff
@@ -0,0 +1,253 @@
+                          rules={{
+                            required: t('required', { value: 'Suurpiiri' }) || '',
+                            validate: {
+                              isPopulated: (d: IOption) => customValidation(d, 'Suurpiiri'),
+                            },
+                          }}
+                          options={locationOptions.districts}
+                        />
+                        <SelectField
+                          {...formProps('division')}
+                          rules={{
+                            required: t('required', { value: 'Kaupunginosa' }) || '',
+                            validate: {
+                              isPopulated: (d: IOption) => customValidation(d, 'Kaupunginosa'),
+                            },
+                          }}
+                          options={locationOptions.divisions}
+                        />
+                        <SelectField
+                          {...formProps('subDivision')}
+                          options={locationOptions.subDivisions}
+                        />
+                      </div>
+                    )}
+
+                    {/* Divider to click */}
+                    <div className="advance-fields-button">
+                      <button onClick={toggleAdvanceFields}>
+                        {t(`groupForm.openAdvanceSearch`)}
+                      </button>
+                      {advanceFieldIcons}
+                    </div>
+                  </div>
+                </form>
+              </div>
+
+              <div>
+                <GroupProjectSearch
+                  getValues={getValues}
+                  control={control}
+                  showAdvanceFields={showAdvanceFields}
+                />
+              </div>
+            </div>
+          </Content>
+
+          <ActionButtons>
+            <Button
+              onClick={handleSubmit(onSubmit)}
+              data-testid="search-projects-button"
+              disabled={!isDirty}
+            >
+              {t('search')}
+            </Button>
+            <Button onClick={handleDialogClose} variant="secondary" data-testid="cancel-search">
+              {t('cancel')}
+            </Button>
+          </ActionButtons>
+        </Dialog>
+      </div>
+    </div>
+  );
+});
+
+DialogContainer.displayName = 'Group Dialog';
+
+const GroupDialog: FC = () => {
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleSetOpen = useCallback(() => setIsOpen((current) => !current), [setIsOpen]);
+  const onOpenGroupForm = useCallback(
+    (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      handleSetOpen();
+    },
+    [handleSetOpen],
+  );
+  const handleClose = useCallback(() => {
+    handleSetOpen();
+  }, [handleSetOpen]);
+  return (
+    <div>
+      {isOpen && <DialogContainer isOpen={isOpen} handleClose={handleClose} />}
```

Comment:
Is it necessary to check isOpen here? Wouldn't it be better to let DialogContainer do the check?

Reply:
DialogContainer contains a hook that performs filtering based on the select fields inside the dialog. One the user opens the dialog and chooses something on the select fields, filtering changes the redux state. If the user then decides to close the container, the hook is not destroyed, the dialog closes. Jonas can better explain this but the workaround was to only render the whole dialog component when isOpen is true, so I added this.